core/state/statedb.go

```diff
@@ -540,13 +544,14 @@ func (s *StateDB) getDeletedStateObject(addr common.Address) *stateObject {
 		if len(enc) == 0 {
 			return nil
 		}
-		if err := rlp.DecodeBytes(enc, &data); err != nil {
+		data = &Account{}
+		if err := rlp.DecodeBytes(enc, data); err != nil {
 			log.Error("Failed to decode state object", "addr", addr, "err", err)
 			return nil
 		}
 	}
 	// Insert into the live set
-	obj := newObject(s, addr, data)
+	obj := newObject(s, addr, *data)
```

Comment:
It feels a bit weird to dereference `data` here. Can't we just pass the pointer and make stateObject store a pointer too?

Reply:
CC @fjl

Reply:
@fjl do you have any comments about this? 
I preferred initially to keep the changes as small as possible, thus didn't change the referencing